django/template/defaultfilters.py

```diff
@@ -151,7 +151,7 @@ def floatformat(text, arg=-1):
             use_l10n = False
             arg = arg[:-1] or -1
     try:
-        input_val = str(text)
+        input_val = repr(text) if isinstance(text, str) else str(text)
```

Comment:
Calling `repr()` on strings will always cause `decimal.InvalidOperation`, IMO this only hides the real issue. What do you think about the following?
```diff
diff --git a/django/template/defaultfilters.py b/django/template/defaultfilters.py
index 23c3a08c67..78881987fc 100644
--- a/django/template/defaultfilters.py
+++ b/django/template/defaultfilters.py
@@ -168,7 +168,7 @@ def floatformat(text, arg=-1):
     except (ValueError, OverflowError, InvalidOperation):
         return input_val
 
-    if not m and p < 0:
+    if not m and p <= 0:
         return mark_safe(
             formats.number_format(
                 "%d" % (int(d)),
```

Reply:
Ah yes, you're right, that should do the trick 👌 
Thanks!